.github/workflows/build.yml

```diff
@@ -1,6 +1,6 @@
 name: Haskell CI
 
-on: [push]
+on: [push, pull_request]
```

Comment:
Right now, this means that in PRs created from branches in this repository, we run the GH Actions checks twice (as in this PR). One solution would be to restrict `push` to certain branches ([docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs)).

Reply:
Indeed that's annoying (that's why we just use `push`). Restricting `push` would also be a bit annoying.  I often create PRs early, and push to the branch, which triggers the CI.  I wonder if this would work:
```yaml
on:
  pull_request:
    types: [opened, reopened, synchronize]
```
But `synchronize` probably doesn't do the same thing as `push`.

Reply:
`cardano-ledger` is using this trick to avoid this:
```
on:
  push:
    branches: [ '*' ]
  pull_request:
    branches: [ "master", "release/*" ]
```
I think it works for PRs that are opened directly from `master` branches of forked repos, but what if one opens a PR from another branch?

Reply:
What about this: https://github.com/input-output-hk/ouroboros-network/pull/4094
I need to check if it will work for a PR from a fork (especially the branch restriction rules).